cpg_pipes/providers/cpg/smdb.py

```diff
@@ -343,16 +350,64 @@ def process_existing_analysis(
                 f'so queueing analysis {label}'
             )
             return None
-    
-    
+
+    def get_ped_file_by_project(self, dataset_name: str, response_type: str):
+        """ Retrieve ped file for a specified SM Project """
+
+        try:
+            families = self.fapi.get_families(dataset_name)
+            family_ids = [family['id'] for family in families]
+            pedigree = self.fapi.get_pedigree(
+                internal_family_ids=family_ids,
+                response_type='json',
+                project=dataset_name,
+                replace_with_participant_external_ids=True,
+            )
+
+            pid_sid_map = self._get_pid_sid_map(dataset_name)
+
+            # Replace External Participant IDs with Sample IDs.
+            # Assumes 1:1 Relationship.
+            for family in pedigree:
+                if family['individual_id'] != '':
+                    family['individual_id'] = pid_sid_map[family['individual_id']]
+
+                if family['maternal_id'] != '':
+                    family['maternal_id'] = pid_sid_map[family['maternal_id']]
+
+                if family['paternal_id'] != '':
+                    family['paternal_id'] = pid_sid_map[family['paternal_id']]
+
+            return pedigree
+        except ApiException:
```

Comment:
Should we fail silently? In what scenarios do we expect SMDB to not work properly?

Reply:
I've thought a bit about this and can't think of a specific case where this would fail that we could anticipate/catch. For example, dataset being invalid would be caught far earlier. And if there aren't corresponding families or pedigree information, then that doesn't constitute a failure (it should populate it with nothing, which it does). 
I've removed the try-catch for now. Do you think there is a better solution here?

Reply:
Removing it is good, I think. If we hit an error, we should look into fixing sample-metadata, instead of failing silently